src/NzbDrone.Core/Movies/AlternativeTitles/AlternativeTitleService.cs

```diff
@@ -63,11 +64,26 @@ public void RemoveTitle(AlternativeTitle title)
             _titleRepo.Delete(title);
         }
 
-        public void DeleteNotEnoughVotes(List<AlternativeTitle> mappingsTitles)
+        public void UpdateTitles(List<AlternativeTitle> titles, Movie movie)
         {
-            var toRemove = mappingsTitles.Where(t => t.SourceType == SourceType.Mappings && t.Votes < 4);
-            var realT = _titleRepo.FindBySourceIds(toRemove.Select(t => t.SourceId).ToList());
-            _titleRepo.DeleteMany(realT);
+            int movieId = movie.Id;
+            // First update the movie ids so we can correlate them later.
+            titles.ForEach(t => t.MovieId = movieId);
+            // Then make sure none of them are the same as the main title.
+            titles = titles.Where(t => t.CleanTitle != movie.CleanTitle).ToList();
+            // Then make sure they are all distinct titles
+            titles = titles.DistinctBy(t => t.CleanTitle).ToList();
+
+            // Now find titles to delete, update and insert.
+            var existingTitles = _titleRepo.FindByMovieId(movieId);
+
+            var insert = titles.Where(t => !existingTitles.Contains(t));
+            var update = titles.Where(t => existingTitles.Contains(t));
```

Comment:
This should be reversed to `existingTitles.Where(t => titles.Contains(t));`. If not then the update fails because the titles list does not have the existing DB ids associated with it. Test doesn't catch it since the UpdateMany is run, but fails at the DB level.

Reply:
True, will update the test as well.